tests/PHPStan/Analyser/data/bug-5336.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Bug5336;
+
+use function PHPStan\Testing\assertType;
+
+interface Stub
+{
+}
+
+interface ProxyQueryInterface
+{
+}
+
+/**
+ * @phpstan-template T of ProxyQueryInterface
+ */
+class Pager
+{
+	/**
+	 * @var T
+	 */
+	private $query;
+
+	/**
+	 * @phpstan-param T $query
+	 */
+	public function __construct(ProxyQueryInterface $query) {
+		$this->query = $query;
+	}
+}
+
+abstract class Test
+{
+	/**
+	 * @var Pager<ProxyQueryInterface&Stub>
+	 */
+	private $pager;
+
+	/**
+	 * @template T of object
+	 * @param class-string<T> $originalClassName
+	 * @return T&Stub
+	 */
+	abstract public function createStub(string $originalClassName): Stub;
+
+	public function sayHello(): void
+	{
+		$query = $this->createStub(ProxyQueryInterface::class);
+		$this->pager = new Pager($query);
+		assertType('Bug5336\Pager<Bug5336\ProxyQueryInterface&Bug5336\Stub>', $this->pager);
```

Comment:
1) Please just do `assertType('Bug5336\Pager<Bug5336\ProxyQueryInterface&Bug5336\Stub>', new Pager($query));`. Assigning `new` to properties has some extra behaviour that might hide bugs here.
2) Please create another test for TypesAssignedToPropertiesRuleTest that actually shows that assignment to property no longer reports an error. If you're gonna use classes with the same names, change the namespace name slightly, like `Bug5336Rule` or something.